packages/calcite-components/src/components/swatch/swatch.scss

```diff
@@ -0,0 +1,122 @@
+//
+// Internal CSS custom properties for component use only. Overwriting is not recommended.
+//
+// --calcite-internal-swatch-inset
+// --calcite-internal-swatch-size
+
+:host {
+  @apply block;
+  --calcite-internal-swatch-inset: var(--calcite-spacing-xxs);
+}
+
+:host([scale="s"]) {
+  .container {
+    --calcite-internal-swatch-size: var(--calcite-spacing-xl);
+  }
+}
+
+:host([scale="m"]) {
+  .container {
+    --calcite-internal-swatch-size: var(--calcite-spacing-xxl);
+  }
+}
+
+:host([scale="l"]) {
+  .container {
+    --calcite-internal-swatch-size: var(--calcite-spacing-xxxl);
+  }
+}
+
+.container {
+  @apply relative focus-base justify-center box-border  overflow-hidden;
+  font-size: var(--calcite-internal-swatch-font-size, var(--calcite-font-size));
+  block-size: var(--calcite-internal-swatch-size, auto);
+  inline-size: var(--calcite-internal-swatch-size, auto);
+  min-inline-size: var(--calcite-internal-swatch-size, auto);
+  border-radius: var(--calcite-swatch-corner-radius, 0);
+
+  &:not(.non-interactive):hover {
+    @apply cursor-pointer;
+  }
+  &:not(.non-interactive):hover {
+    box-shadow: 0 0 0 2px var(--calcite-color-border-2);
+  }
+  &.selectable {
+    @apply cursor-pointer;
+  }
+  &:not(.non-interactive):focus {
+    @apply focus-normal;
+  }
+}
+
+.swatch {
+  position: absolute;
+  @apply inline-flex overflow-hidden absolute z-10;
+  block-size: var(--calcite-internal-swatch-size, auto);
+  inline-size: var(--calcite-internal-swatch-size, auto);
+  min-inline-size: var(--calcite-internal-swatch-size, auto);
+  border-radius: var(--calcite-swatch-corner-radius, 0);
+}
+
+:host([selected]) .swatch {
+  inset: var(--calcite-internal-swatch-inset);
+  block-size: calc(var(--calcite-internal-swatch-size, auto) - calc(2 * var(--calcite-internal-swatch-inset)));
+  inline-size: calc(var(--calcite-internal-swatch-size, auto) - calc(2 * var(--calcite-internal-swatch-inset)));
+  min-inline-size: calc(var(--calcite-internal-swatch-size, auto) - calc(2 * var(--calcite-internal-swatch-inset)));
+}
+
+:host([selected]) .container {
+  box-shadow:
+    inset 0 0 0 2px var(--calcite-color-text-1),
+    inset 0 0 0 4px var(--calcite-color-foreground-1);
+}
+
+:host([selected]) .image-container {
+  inset: var(--calcite-internal-swatch-inset);
+}
+
+.image-container {
+  @apply inline-flex overflow-hidden absolute z-20;
+  inset: var(--calcite-spacing-px);
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  pointer-events: none;
+  border-radius: var(--calcite-swatch-corner-radius, 0);
+}
+
+.empty-container {
+  @apply absolute inset-0 flex items-center justify-center z-30;
+}
+
+.disabled-container {
+  @apply absolute inset-0 flex items-center justify-center z-30;
+}
+
+.swatch {
+  @apply overflow-hidden;
+  block-size: inherit;
+  inline-size: inherit;
+}
+
+.swatch--no-color {
+  rect {
+    fill: var(--calcite-color-foreground-1);
+  }
+}
+
+.checker {
+  fill: #cacaca;
```

Comment:
should this color use a css var?

Reply:
The closest core token is `var(--calcite-color-neutral-blk-050)`